pkg/api/meta/restmapper.go

```diff
@@ -169,6 +169,15 @@ func (m *DefaultRESTMapper) ResourceSingularizer(resource string) (singular stri
 
 // VersionAndKindForResource implements RESTMapper
 func (m *DefaultRESTMapper) VersionAndKindForResource(resource string) (defaultVersion, kind string, err error) {
+	if parts := strings.Split(resource, "/"); len(parts) > 1 {
+		if len(parts) > 2 {
+			return "", "", fmt.Errorf("in version and kind for resource, resource %q has an invalid name", resource)
```

Comment:
Is "in version and kind for resource," any helpful for an end-user? If we already display the rest of the error message elsewhere how about a smaller prefix?